cmd/ears/run.go

```diff
@@ -0,0 +1,85 @@
+package main
+
+import (
+	"fmt"
+	"github.com/rs/zerolog"
+	"os"
+
+	"github.com/rs/zerolog/log"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"github.com/xmidt-org/ears/internal/pkg/app"
+	"github.com/xmidt-org/ears/internal/pkg/cli"
+	"go.uber.org/fx"
+)
+
+var runCmd = &cobra.Command{
+	Use:   "run",
+	Short: "Runs the EARS microservice",
+	Long:  `Runs the EARS microservice`,
+	Run: func(cmd *cobra.Command, args []string) {
+
+		logger, err := InitGlobalLogger()
+		if err != nil {
+			log.Logger.Error().Str("op", "InitLogger").Msg(err.Error())
+			os.Exit(1)
+		}
+		earsApp := fx.New(
+			fx.Provide(
+				ViperConfig,
+				app.NewLogger,
+				app.NewRouter,
+				app.NewMiddlewares,
+				app.NewMux,
+			),
+			fx.Logger(logger),
+			fx.Invoke(app.SetupAPIServer),
+		)
+		earsApp.Run()
+	},
+}
+
+func ViperConfig() app.Config {
+	return viper.Sub("ears")
+}
+
+//Initialize logging. We are initializing the zerolog global logger for uber/fx logging
+//The application itself will not use the global logger. Instead, it use the logger returned by
+//app.NewLogger function, which is a child logger of the global logger
+func InitGlobalLogger() (fx.Printer, error) {
```

Comment:
InitGlobalLogger should take in an `app.Config`.  Otherwise, you get viper tangled throughout the code.  The added benefit is also you just do `config.GetString("logLevel")`